homeassistant/components/template/template_entity.py

```diff
@@ -338,16 +338,23 @@ def _handle_results(
                 )
             return
 
+        errors = []
         for update in updates:
             for template_attr in self._template_attrs[update.template]:
                 template_attr.handle_result(
                     event, update.template, update.last_result, update.result
                 )
+                if isinstance(update.result, TemplateError):
+                    errors.append(update.result)
 
         if not self._preview_callback:
             self.async_write_ha_state()
             return
 
+        if errors:
+            self._preview_callback(None, None, None, str(errors[-1]))
```

Comment:
Why do we only pick the last error? Should we allow propagating a list of errors instead?